src/Config.js

```diff
@@ -80,15 +81,62 @@ module.exports = class extends ChainedMap {
 
         switch (key) {
           case 'node':
-          case 'output':
           case 'resolve':
           case 'resolveLoader':
           case 'devServer':
           case 'module': {
             return this[key].merge(value);
           }
 
+          /**
+           * @example:
+           *
+           * input:
+           * '/code/example/dist/[name].js'
+           *
+           * output:
+           * {
+           *   path: '/code/example/dist',
+           *   filename: '[name].js',
+           * }
+           *
+           *
+           */
+          case 'output': {
+            if (typeof value === 'string') {
+              const filename = path.basename(value);
+              const dir = path.dirname(value);
+              const asObject = {
+                path: dir,
+                filename,
+              };
+
+              return this.output.merge(asObject);
+            }
+          }
+
+          /**
+           * take a string entry,
+           * file name becomes property,
+           * value becomes an array,
+           * if the same prop exists,
+           * it is merged in with existing values
+           *
+           * @example:
+           *
+           * input: './src/front/index.js'
+           * output: {index: ['./src/front/index.js']}
+           */
           case 'entry': {
+            if (typeof value === 'string') {
```

Comment:
I can't understand the rationale for this one. Why should `'./src/front/index.js'` map to an entry point named `index`? Many directories could contain a file named `index.js`, without the justification that they should be bundled together. It seems the using the parent directory name would be a better fit, e.g. `front` entry. Making this conventional I think will have unintended consequences.

Just need to understand why we should go this route as opposed to using a different name (or whether accepting a string at all here is a good idea).